pkg/client/unversioned/clientcmd/api/types.go

```diff
@@ -92,6 +92,8 @@ type AuthInfo struct {
 	Username string `json:"username,omitempty"`
 	// Password is the password for basic authentication to the kubernetes cluster.
 	Password string `json:"password,omitempty"`
+	// AuthPlugin is a custom plugin used to authenticate to the kubernetes cluster.
+	AuthPlugin string `json:"auth-plugin,omitempty"`
```

Comment:
This sounds pretty flexible, and at first glace I really like this. Upon
further reflection I wonder if a configmap might be more appropriate? Or
would that be too restrictive? Configmap is an API standard after all..

On Mon, Mar 21, 2016 at 7:27 PM, CJ Cullen <notifications@github.com> wrote:

> In pkg/client/unversioned/clientcmd/api/types.go
> <https://github.com/kubernetes/kubernetes/pull/23066#discussion_r56915022>
> :
>
> > @@ -92,6 +92,8 @@ type AuthInfo struct {
> >  	Username string `json:"username,omitempty"`
> >  	// Password is the password for basic authentication to the kubernetes cluster.
> >  	Password string `json:"password,omitempty"`
> > +	// AuthPlugin is a custom plugin used to authenticate to the kubernetes cluster.
> > +	AuthPlugin string `json:"auth-plugin,omitempty"`
>
> Okay. How about something like:
>
> type AuthProviderConfig struct {
>     Name string
>     Config json.RawMessage
> }
>
> Plugins would then be provided with the specified Config when
> instantiated. Simple plugins (like mine) can specify a Name, and leave
> Config blank. More complex plugins could define and parse their own config,
> and it would all look nice and pretty in the kubeconfig file.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23066/files/759f65de42125a402f759a8080b0100e8cd7cc9e#r56915022>
>